compiler/ast/ast.go

```diff
@@ -83,6 +83,19 @@ type Cast struct {
 	Type Expr   `json:"type"`
 }
 
+type IndexExpr struct {
+	Kind  string `json:"kind" unpack:""`
+	Expr  Expr   `json:"expr"`
+	Index Expr   `json:"index"`
+}
+
+type SliceExpr struct {
+	Kind string `json:"kind" unpack:""`
+	Expr Expr   `json:"expr"`
+	Low  Expr   `json:"low"`
+	High Expr   `json:"high"`
```

Comment:
runtime/sam/expr.Slice and compiler/kernel.Builder.compileSlice (and maybe other functions) call these "from" and "to". I don't have any preference between those names (though I do have my usual, mild preference for keeping the diff small) but let's stay consistent.

Reply:
The go std library uses high low but lets just keep it from/to.